src/plugins/embeddable/public/react_embeddable_system/react_embeddable_renderer.tsx

```diff
@@ -69,6 +69,7 @@ export const ReactEmbeddableRenderer = <
     | 'hideLoader'
```

Comment:
I see one change that is probably not the most user-friendly. When clicking on a contextual `edit filters` button, it opens the inline editor. Expected (previous) behavior would be to open a full Lens editor. It was a better behavior because currently user cannot modify filters within the inline editor, so it can be confusing for them having to go to the Lens editor from there. 
 

<img width="471" alt="Screenshot 2024-11-22 at 14 50 53" src="https://github.com/user-attachments/assets/2bf0495c-30ba-46cc-9b8c-c37c8406b2bb">

Reply:
Good catch. On it...

Reply:
I'm managed to narrow down the issue here, which is due to the new `onEdit` architecture that gives priority to inline editing vs full editor.
The new `onEdit` API doesn't accept any argument at the moment and that prevents Lens to communicate the specific context: introducing a way to communicate the Lens embeddable what is the preferred context to "edit" would allow to prompt different edit experiences based on the context.
I have a fix but I would prefer to defer it after this PR into a follow up and after a discussion with the @elastic/kibana-presentation team about it.

Reply:
It is ok to postpone the fix to this one to a separate PR 🆗